src/Core/Content/ProductExport/Service/ProductExportFileHandler.php

```diff
@@ -59,8 +59,13 @@ public function isValidFile(string $filePath, ExportBehavior $behavior, ProductE
         if (!$this->fileSystem->has($filePath)) {
             return false;
         }
+        
+        // Regenerate always if cache is expired. Even if product export was generated by cronjob.
+        if($this->isCacheExpired($behavior, $productExport)) {
```

Comment:
Could you add a test and a changelog file?

Reply:
@shyim i can implement a changelog. I'm not sure how to write a test for this.

Reply:
Hey sorry for the late response. I was in vacation.

Create a new class ProductExportFileHandlerTest and use the IntegreationTestBehaviour.

With `$this->getContainer()->get(ProductExporter::class)` you get the exporter service and you can call then in the test methods the public method and assert what you should